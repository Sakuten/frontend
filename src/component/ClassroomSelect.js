import React from 'react'
import { observer } from 'mobx-react'

const ClassroomSelect = ({list, value, onChange}) => (
  <select data-test='classroom-select' name='classrooms' value={value} onChange={e => onChange(e.target.value)}>
    {
      list.map(c =>
        <option data-test='classroom-option' key={c.id} value={c.id}>{c.grade} {c.name}</option>
      )
    }
  </select>
)

export default observer(ClassroomSelect)
