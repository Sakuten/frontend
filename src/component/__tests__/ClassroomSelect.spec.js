import React from 'react'
import { shallow } from 'enzyme'
import ClassroomSelect from '../ClassroomSelect'

const setup = propOverrides => {
  const props = Object.assign({
    list: [
      {
        'grade': 5,
        'id': 1,
        'index': 0,
        'name': 'A'
      },
      {
        'grade': 5,
        'id': 2,
        'index': 1,
        'name': 'B'
      }
    ],
    value: 0,
    onChange: jest.fn()
  }, propOverrides)

  const wrapper = shallow(<ClassroomSelect {...props} />)

  return {
    props,
    wrapper
  }
}

describe('components', () => {
  describe('ClassroomSelect', () => {
    it('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders two options', () => {
      const { wrapper } = setup()
      expect(wrapper.find('[data-test="classroom-option"]').length).toBe(2)
    })

    it('renders with grade and name', () => {
      const { wrapper } = setup()
      expect(wrapper.find('[data-test="classroom-option"]').at(0).text()).toBe('5 A')
      expect(wrapper.find('[data-test="classroom-option"]').at(1).text()).toBe('5 B')
    })

    it('calls onChange with id when something is selected', () => {
      const mock = jest.fn()
      const { wrapper } = setup({onChange: mock})
      wrapper.find('[data-test="classroom-select"]').simulate('change', {target: {value: 2}})
      expect(mock).toBeCalledWith(2)
    })

    it('renders specified id', () => {
      const { wrapper } = setup({value: 2})
      expect(wrapper.find('[data-test="classroom-select"]').props().value).toBe(2)
    })
  })
})
