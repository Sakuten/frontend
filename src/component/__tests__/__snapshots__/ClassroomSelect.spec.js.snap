// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ClassroomSelect render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClassroomSelect
    list={
      Array [
        Object {
          "grade": 5,
          "id": 1,
          "index": 0,
          "name": "A",
        },
        Object {
          "grade": 5,
          "id": 2,
          "index": 1,
          "name": "B",
        },
      ]
    }
    onChange={[MockFunction]}
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          data-test="classroom-option"
          value={1}
        >
          5
           
          A
        </option>,
        <option
          data-test="classroom-option"
          value={2}
        >
          5
           
          B
        </option>,
      ],
      "data-test": "classroom-select",
      "name": "classrooms",
      "onChange": [Function],
      "value": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            5,
            " ",
            "A",
          ],
          "data-test": "classroom-option",
          "value": 1,
        },
        "ref": null,
        "rendered": Array [
          5,
          " ",
          "A",
        ],
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            5,
            " ",
            "B",
          ],
          "data-test": "classroom-option",
          "value": 2,
        },
        "ref": null,
        "rendered": Array [
          5,
          " ",
          "B",
        ],
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            data-test="classroom-option"
            value={1}
          >
            5
             
            A
          </option>,
          <option
            data-test="classroom-option"
            value={2}
          >
            5
             
            B
          </option>,
        ],
        "data-test": "classroom-select",
        "name": "classrooms",
        "onChange": [Function],
        "value": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              5,
              " ",
              "A",
            ],
            "data-test": "classroom-option",
            "value": 1,
          },
          "ref": null,
          "rendered": Array [
            5,
            " ",
            "A",
          ],
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              5,
              " ",
              "B",
            ],
            "data-test": "classroom-option",
            "value": 2,
          },
          "ref": null,
          "rendered": Array [
            5,
            " ",
            "B",
          ],
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
